r/tests/testthat/test-dplyr-summarize.R

```diff
@@ -836,28 +835,68 @@ test_that("Expressions on aggregations", {
   expect_warning(
     record_batch(tbl) %>% summarise(any(any(lgl))),
     paste(
-      "Aggregate within aggregate expression",
-      "any\\(any\\(lgl\\)\\) not supported in Arrow"
+      "In any\\(any\\(lgl\\)\\), aggregate within aggregate expression",
+      "not supported in Arrow"
     )
   )
 
   # Check aggregates on aggregates with more complex calls
   expect_warning(
     record_batch(tbl) %>% summarise(any(any(!lgl))),
     paste(
-      "Aggregate within aggregate expression",
-      "any\\(any\\(!lgl\\)\\) not supported in Arrow"
+      "In any\\(any\\(!lgl\\)\\), aggregate within aggregate expression",
+      "not supported in Arrow"
     )
   )
   expect_warning(
     record_batch(tbl) %>% summarise(!any(any(lgl))),
     paste(
-      "Aggregate within aggregate expression",
-      "any\\(any\\(lgl\\)\\) not supported in Arrow"
+      "In \\!any\\(any\\(lgl\\)\\), aggregate within aggregate expression",
+      "not supported in Arrow"
     )
   )
 })
 
+test_that("Weighted mean", {
+  compare_dplyr_binding(
+    .input %>%
+      group_by(some_grouping) %>%
+      summarize(
+        weighted_mean = sum(int * dbl) / sum(dbl)
+      ) %>%
+      collect(),
+    tbl
+  )
+
+  division <- function(x, y) x / y
+  compare_dplyr_binding(
+    .input %>%
+      group_by(some_grouping) %>%
+      summarize(
+        weighted_mean = division(sum(int * dbl), sum(dbl))
+      ) %>%
+      collect(),
+    tbl
+  )
+
+  # We can also define functions that call supported aggregation functions
+  # and it just works
+  wtd_mean <- function(x, w) sum(x * w) / sum(w)
+  withr::local_options(list(arrow.debug = TRUE))
```

Comment:
Nice, this is a helpful catch / test that honestly I could see being helpful when debugging this too — but makes this super clear that it's hitting this code and not some other path